engine/src/main/java/org/terasology/particles/components/ParticleSystemComponent.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.particles.components;
+
+import org.lwjgl.util.vector.Vector2f;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.Owns;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.particles.events.ParticleSystemUpdateEvent;
+import org.terasology.particles.rendering.ParticleRenderer;
+import org.terasology.rendering.assets.texture.Texture;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Represents a particle system. A particle system must have emitters in order to produce particles.
+ */
+public class ParticleSystemComponent implements Component {
+
+    /**
+     * Reference to the entity this component is attached to
+     */
+    public EntityRef ownerEntity;
+
+    /**
+     * The renderer that should be used to render this system.
+     */
+    private Class<ParticleRenderer> renderer;
+
+    /**
+     * The lifetime of this system
+     */
+    public float maxLifeTime = Float.POSITIVE_INFINITY;
+
+    /**
+     * Destroy this particle system when its last emitter is destroyed TODO: Unimplemented
+     */
+    public boolean destroyWhenFinished = true;
+
+    /**
+     * The amount of particles in this system's pool. Emitters under this system will share the particles.
+     */
+    public int particlePoolSize = 1000;
+
+    @Owns
+    private List<EntityRef> emitters = new LinkedList<>();
+
+    @Owns
+    private List<EntityRef> affectors = new LinkedList<>();
```

Comment:
hmm I thought emitters and affectors would each be represented by a component.

Since those fields are just implementation details I suggest you keep track of them via a map with particle system entity to value map that gets updated when the `ParticleSystemComponent` component gets activated or deactivated. See `OnActivatedComponent` and `BeforeDeactivateComponent`. For an example you can look at `NameTagClientSystem. That way we only get OnChangedComponent events when a actual public component field changes. Everytime a component of a persistent entity gets changed, the component gets copied for the background saving task. So we want to avoid any unnecessary component modifications events. Also it keeps the component classes simple which makes future changes of how we do components easier and helps developers that just want to get a quick overview of a component.